packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -325,6 +340,7 @@ export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget
         if (IBaseTerminalServer.validateId(terminalId)) {
             return terminalId;
         }
+        this.terminalKind = 'user';
```

Comment:
I don't think we should change a value of kind after construction, it should be assigned once and be always readonly.

What was your motivation?

Reply:
the reason behind this line is, when a terminal fails to attach to a process, it becomes a "user terminal". Please see this GIF:

![Peek 2020-03-23 18-03](https://user-images.githubusercontent.com/37082801/77367562-cfc6ee00-6d30-11ea-9d9d-3e6fe3e25174.gif)

Reply:
> the reason behind this line is, when a terminal fails to attach to a process, it becomes a "user terminal".

I don't think it is correct from the lifecycle and responsibility perspective. If the task extensions is responsible for creating a task terminal, then it should be responsible for destroying it or giving up the ownership (i.e. changing a kind of terminal to `user`), not a terminal itself.

Why don't just print `Terminal will be reused for tasks`? It removes new logic with the managing terminal ownership after creation and is more transparent for a user.

Reply:
> Why don't just print `Terminal will be reused for tasks`? It removes new logic with the managing terminal ownership after creation and is more transparent for a user.

this is in the terminal extension. since terminals have no knowledge of task, i guess we probably need to add another event to inform the task extension that this terminal widget fails to attach to a process ...